services/job_subscriber.go

```diff
@@ -71,14 +95,69 @@ func (js *JobSubscriber) Disconnect() {
 }
 
 // OnNewHead resumes all pending job runs based on the new head activity.
-func (js *JobSubscriber) OnNewHead(head *models.BlockHeader) {
-	pendingRuns, err := js.Store.JobRunsWithStatus(models.RunStatusPendingConfirmations)
+func (js *jobSubscriber) OnNewHead(head *models.BlockHeader) {
+	pendingRuns, err := js.Store.JobRunsWithStatus(models.RunStatusPendingConfirmations, models.RunStatusInProgress)
 	if err != nil {
 		logger.Error(err.Error())
 	}
+
+	activeJobRunIDs := make(map[string]struct{})
+
+	js.workerMutex.Lock()
+	defer js.workerMutex.Unlock()
 	for _, jr := range pendingRuns {
-		if _, err := ExecuteRunAtBlock(jr, js.Store, jr.Result, head.ToIndexableBlockNumber()); err != nil {
-			logger.Error(err.Error())
+		activeJobRunIDs[jr.ID] = struct{}{}
+
+		workerChannel := js.WorkerChannelFor(jr)
+		blockNumber := head.ToIndexableBlockNumber()
+		workerChannel <- blockNumber
```

Comment:
To reiterate, I was thinking this could turn into

```
js.Store.RunQueue <- entry{jr.Result, blockNumber}
```

where `workerChannel` nuances are handled in `RunManager`